pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/memory/unsafe/DirectMemory.java

```diff
@@ -56,8 +56,12 @@ public void flush() {
   @Override
   public synchronized void close() {
     if (!_closed) {
-      Unsafer.UNSAFE.freeMemory(_address);
-      _closed = true;
+      synchronized (this) {
```

Comment:
This is not necessary. The whole `close()` is synchronized

Reply:
You are right, but we can remove that first synchronized now, right?

Reply:
This method is almost always invoked only once, so the double-checked locking is not helping much here. Even for singleton class access, I've read that for modern JVM the overhead of locking is negligible, and people are suggesting using global locking for simplicity